static/app/views/dashboards/manage/index.tsx

```diff
@@ -64,12 +65,22 @@ const SORT_OPTIONS: SelectValue<string>[] = [
 ];
 
 const SHOW_TEMPLATES_KEY = 'dashboards-show-templates';
+const LAYOUT_KEY = 'dashboards-overview-layout';
+
+type DashboardsLayout = 'grid' | 'list';
 
 function shouldShowTemplates(): boolean {
   const shouldShow = localStorage.getItem(SHOW_TEMPLATES_KEY);
   return shouldShow === 'true' || shouldShow === null;
 }
 
+function getDashboardsOverviewLayout(): DashboardsLayout {
+  const dashboardsLayout = localStorage.getItem(LAYOUT_KEY);
+  return dashboardsLayout === 'grid' || dashboardsLayout === 'list'
+    ? dashboardsLayout
+    : 'grid';
```

Comment:
We should put `'grid'` and `'list'` in constants at the top of the page to reuse them everywhere instead of dealing with strings